File: .tmp_py4013/batch_12_dump/3417_zigzag_grid_traversal_with_skip.js
// LeetCode 3417 - Zigzag Grid Traversal With Skip
// https://leetcode.com/problems/zigzag-grid-traversal-with-skip/

var zigzagTraversal = function(grid) {
    const ans = [];
    let skip = false;
    for (let i = 0; i < grid.length; i++) {
        const row = grid[i];
        if (i % 2 === 0) {
            for (const v of row) {
                if (!skip) ans.push(v);
                skip = !skip;
            }
        } else {
            for (let j = row.length - 1; j >= 0; j--) {
                if (!skip) ans.push(row[j]);
                skip = !skip;
            }
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check17/3417_zigzag_grid_traversal_with_skip.rs
struct Solution;
// LeetCode 3417 - Zigzag Grid Traversal With Skip
// https://leetcode.com/problems/zigzag-grid-traversal-with-skip/

impl Solution {
    pub fn zigzag_traversal(grid: Vec<Vec<i32>>) -> Vec<i32> {
        let mut ans = Vec::new();
        let mut skip = false;
        for (i, row) in grid.iter().enumerate() {
            if i % 2 == 0 {
                for &v in row {
                    if !skip {
                        ans.push(v);
                    }
                    skip = !skip;
                }
            } else {
                for j in (0..row.len()).rev() {
                    if !skip {
                        ans.push(row[j]);
                    }
                    skip = !skip;
                }
            }
        }
        ans
    }
}

fn main() {}
